### PR TITLE
(maint) If consul is enabled, register puppetdb service on startup

### PR DESCRIPTION
**Pull request by underscorgan**

### CODE REVIEW

**Comment by Iristyle** [comment]
Retargeted at `master` so that we don't forget in the future.

**Comment by underscorgan** [comment]
This should be good to go!

File: docker/puppetdb/docker-entrypoint.d/30-consul.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+if [ "$CONSUL_ENABLED" = "true" ]; then
+  ipaddress="$(ifconfig eth0 | grep 'inet addr' | cut -d ':' -f 2 | cut -d ' ' -f 1)"
```
Comment (nwolfe) [inline docker/puppetdb/docker-entrypoint.d/30-consul.sh]: I was wondering about the availability of `ifconfig`, and noticed that it does not exist in the `ubuntu` and `centos` images. It does exist in `alpine` though.

So something else that we install in these images causes this to be brought in?

Just wanted to point it out, not sure there's anything to do about it though.

Comment (nwolfe) [reply docker/puppetdb/docker-entrypoint.d/30-consul.sh]: Looks like you get `ifconfig` via the `net-tools` package on Ubuntu.

**Comment by Iristyle** [comment]
Retriggered PDB container tests (since earlier cells failed) and it passes.

File: docker/puppetdb/docker-entrypoint.d/30-consul.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  consul_hostname="${CONSUL_HOSTNAME:-consul}"
```
Comment (Iristyle) [inline docker/puppetdb/docker-entrypoint.d/30-consul.sh]: In the puppetserver PR, we made these values `CONSULT_HOSTNAME` and `CONSUL_PORT` for consistency.

Comment (underscorgan) [reply docker/puppetdb/docker-entrypoint.d/30-consul.sh]: @Iristyle updated

**Comment by Iristyle** [comment]
Kicked again to run the container tests. Still green - merging!

**Pull request merged**